File: manual/data/python/tests/test_networkx_diameter.py
import unittest
from function_networkx_diameter import GraphUtils

class TestCalculateDiameter(unittest.TestCase):

    def test_success(self):
        result = GraphUtils().calculate_diameter("A-B,B-C,C-D,D-E")
        self.assertTrue(result.startswith("success"))

    def test_diameter_value(self):
        result = GraphUtils().calculate_diameter("A-B,B-C,C-D,D-E")
        if result.startswith("success"):
            diameter = int(result.split(":")[1].strip())
            self.assertEqual(diameter, 4)

    def test_non_numeric(self):
        result = GraphUtils().calculate_diameter("A-B,B-C,C-D,D-E")
        self.assertIsInstance(result, str)

    def test_disconnected_graph(self):
        result = GraphUtils().calculate_diameter("A-B,C-D")
        self.assertEqual(result, "failed")

    def test_empty_graph(self):
        result = GraphUtils().calculate_diameter("")
        self.assertEqual(result, "failed")

if __name__ == '__main__':
    unittest.main()

File: manual/data/java/tests/test_networkx_diameter.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_networkx_diameter {
    @Test
    public void test_success() {
        String result = new function_networkx_diameter()
                .calculate_diameter("A-B,B-C,C-D,D-E");
        assertTrue(result.startsWith("success"));
    }

    @Test
    public void test_diameter_value() {
        String result = new function_networkx_diameter()
                .calculate_diameter("A-B,B-C,C-D,D-E");
        if (result.startsWith("success")) {
            int diameter = Integer.parseInt(result.split(":")[1].trim());
            assertEquals(4, diameter);
        }
    }

    @Test
    public void test_non_numeric() {
        String result = new function_networkx_diameter()
                .calculate_diameter("A-B,B-C,C-D,D-E");
        assertInstanceOf(String.class, result);
    }

    @Test
    public void test_disconnected_graph() {
        String result = new function_networkx_diameter()
                .calculate_diameter("A-B,C-D");
        assertEquals("failed", result);
    }

    @Test
    public void test_empty_graph() {
        String result = new function_networkx_diameter()
                .calculate_diameter("");
        assertEquals("failed", result);
    }
}